### ISSUE TITLE
snowflake_failover_group object types showing changes on every run

### ISSUE THREAD

**Issue by gdelia-pm** [Snowflake-Labs/terraform-provider-snowflake#2517]
### Terraform CLI and Provider Versions

```
 .\terraform.exe version
Terraform v1.7.3
on windows_amd64
+ provider registry.terraform.io/snowflake-labs/snowflake v0.86.0
```

### Terraform Configuration

```terraform
resource "snowflake_failover_group" "SOURCE_FAILOVER_GROUP" {
  depends_on = [snowflake_database.DB, snowflake_storage_integration.AZURE_STORAGE_INTEGRATION_DEV, snowflake_role.FUNCTIONAL_ROLES, snowflake_warehouse.WH,]

  name                      = "FAILOVER_GROUP1"
  object_types              = ["ACCOUNT PARAMETERS", "DATABASES", "INTEGRATIONS", "NETWORK POLICIES", "RESOURCE MONITORS", "ROLES", "USERS", "WAREHOUSES"]
  allowed_accounts          = ["ACCOUNT.CORE1", "ACCOUNT.CORE_DR1"]
  allowed_integration_types = ["SECURITY INTEGRATIONS", "STORAGE INTEGRATIONS"]
  allowed_databases         = toset([for db in snowflake_database.DB : db.name])

  replication_schedule {
    interval = 15 # minutes
  }
}
```

### Expected Behavior

After the initial apply no further plan should be generated.  

### Actual Behavior

On every plan, object type PARAMETERS is detected as removed, while ACCOUNT PARAMETERS is set to be added.
This seems to be a result of how snowflake stores the account parameter option, as it's noted as just parameter when running show failover groups. I think the tf provider is reading the parameters on each load and seeing that as different than the accepted object type of account parameters. 

I can work around this by creating the group with the "ACCOUNT PARAMETERS" object type, applying the config, then changing my object type to "PARAMETERS" to suppress the change, but if I have to make another change that breaks things. 

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # snowflake_failover_group.SOURCE_FAILOVER_GROUP will be updated in-place
  ~ resource "snowflake_failover_group" "SOURCE_FAILOVER_GROUP" {
        id                        = "FAILOVER_GROUP1"
        name                      = "FAILOVER_GROUP1"
      ~ object_types              = [
          - "PARAMETERS",
          + "ACCOUNT PARAMETERS",
            # (7 unchanged elements hidden)
        ]
        # (5 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. add the failover resource with object type "ACCOUNT PARAMETERS"
2. `terraform apply`
3. `terraform apply` will show an unexpected change
4. changing the object type on the failover resource to "PARAMETERS" will stop detecting a new change
5. `terraform destroy`
6. `terraform apply` will fail with the PARAMETERS object type set for a new resource

### How much impact is this issue causing?

Medium

**Comment by sfc-gh-asawicki** [comment]
Hey @gdelia-pm. Thanks for reaching out to us.

I confirmed that this is undocumented Snowflake behavior (returning `PARAMETERS` instead of `ACCOUNT PARAMETERS`). I have adjusted the SDK logic in #2522. It should be merged before 0.87.0 release.

**Comment by sfc-gh-asawicki** [comment]
Hey @gdelia-pm . We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!

**Comment by gdelia-pm** [comment]
Confirming this is fixed in 0.87.  Thanks!
